pkg/proxy/healthcheck/healthcheck_test.go

```diff
@@ -411,3 +424,51 @@ func testHealthzHandler(server httpServer, status int, t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestServerWithSelectiveListeningAddress(t *testing.T) {
+	listener := newFakeListener()
+	httpFactory := newFakeHTTPServerFactory()
+
+	// limiting addresses to loop back. We don't want any cleverness here around getting IP for
+	// machine nor testing ipv6 || ipv4. using loop back guarantees the test will work on any machine
```

Comment:
are not fake the listeners?

Reply:
They are. They also have addresses handling logic that mimics the real thing.

Reply:
Then we can use non loopback addresses, the comment is not correct , no?